functions/Find-DbaOrphanedFile.ps1

```diff
@@ -0,0 +1,231 @@
+			
+	        foreach ($db in $databaselist)
+	        {
+		        # Add support for Full Text Catalogs in Sql Server 2005 and below
+		        if ($server.VersionMajor -lt 10)
+		        {
+			        #$dbname = $db.name
+			        $fttable = $null = $smoserver.Databases[$database].ExecuteWithResults('sp_help_fulltext_catalogs')
+					
+			        foreach ($ftc in $fttable.Tables[0].rows)
+			        {
+				        $name = $ftc.name
+				        $physical = $ftc.Path
+				        $logical = "sysft_$name"
+				        $null = $ftfiletable.Rows.add($database, "FULLTEXT", $logical, $physical)
+			        }
+		        }
+	        }
+			
+	        $null = $dbfiletable.Tables.Add($ftfiletable)
+	        return $dbfiletable
+        }
+
+	}
+	
+	PROCESS
+	{
+        # Get the servername from the sqlinstance parameter
+        if($SqlServer.Contains('\'))
+        {
+            $servername = $SqlServer.Split('\')[0]
+        }
+        else
+        {
+            $servername = $SqlServer
+        }
+
+        # Check if the servername is the local server
+        if($env:computername -eq $servername)
+        {
+            # Check if the user is executing the function in elevated mode
+            if(-not ((New-Object Security.Principal.WindowsPrincipal ([Security.Principal.WindowsIdentity]::GetCurrent())).IsInRole([Security.Principal.WindowsBuiltinRole]::Administrator))) 
+            {
+	            Write-Host "You must run Windows PowerShell as Administrator - Elevated Mode"
+	            Throw
+            }
+
+            # Set the method to locally
+            $method = 'local'
+        }
+        else
+        {
+            # Set the method to remote
+            $method = 'remote'
+        }
+
+		#Write-Output "Attempting to connect to SQL Server.."
+		
+		$sourceserver = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+
+        # Get all the database files
+        $databasefiles = Get-SqlFileStructure -server $sourceserver
+
+        # Check if there are any database files retrieved
+        if($databasefiles.count -ge 1)
+        {
+            # Create the orphaned files variable
+            $orphanedfiles = @() 
+
+            # Getthe default data and log directories from the instance
+            $SourceDirectory += $sourceserver.RootDirectory + "\DATA"
+
+            # Check if the default file path is set
+            if (($sourceserver.Settings.DefaultFile).Length -eq 0) 
+            {
+	            $SourceDirectory += $sourceserver.Information.MasterDBPath
+	        }
+            else
+            {
+                $SourceDirectory += $sourceserver.Settings.DefaultFile
+            }
+
+            # Check if the default log path is set
+	        if (($sourceserver.Settings.DefaultLog).Length -eq 0) {
+	            $SourceDirectory = $sourceserver.Information.MasterDBLogPath
+	        }
+            else
+            {
+                $SourceDirectory += $sourceserver.Settings.DefaultLog

```

Comment:
There is an internal command in SharedFunctions.ps1 to get this because it's not always set properly in SMO. Please see Get-SqlDefaultPaths, which tries 3 different ways to get defaults.